src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java

```diff
@@ -1188,15 +1203,21 @@ public InMemoryOutput downloadOutputs(RemoteAction action, RemoteActionResult re
         }
 
         if (isInMemoryOutputFile) {
-          downloadsBuilder.add(
-              transform(
-                  remoteCache.downloadBlob(
-                      context, inMemoryOutputPath.getPathString(), file.digest()),
-                  data -> {
-                    inMemoryOutputData.set(data);
-                    return null;
-                  },
-                  directExecutor()));
+          // There is no way to distinguish between an empty file and a file that has not been
+          // inlined.
```

Comment:
That's quite a footgun in the protocol!

Reply:
It's actually not, I missed that we can just check `digest.size_bytes`. Added this.